app/models/webhook.rb

```diff
@@ -1,11 +1,17 @@
 class Webhook < ActiveRecord::Base
   has_soft_deletion default_scope: true
-  validates :branch, uniqueness: { scope: [ :stage], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
+  validates :branch, uniqueness: { scope: [ :stage ], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
 
   belongs_to :project
   belongs_to :stage
 
+  SOURCES = %w[semaphore tddium travis jenkins buildkite github].freeze
```

Comment:
Tried `SOURCES = Integrations::BaseController.subclasses.map(&:service_name).freeze` but had trouble because the subclasses hadn't been loaded when the constant was evaluated (base controller was probably autoloaded then).